misc/generate-routes-file-for-dynamic-routes.js

```diff
@@ -0,0 +1,120 @@
+* This js script also generates a mapper `case-sensitive-video-id-mapper.js` located in `src/functions/utils`
+* To map lower case video ids -> case sensitive video Ids
+* https://answers.netlify.com/t/my-url-paths-are-forced-into-lowercase/1659 */
+const fetch = require("node-fetch");
+const {ensureFileSync, writeFileSync} = require('fs-extra');
+const DIST_DYNAMIC_ROUTES_FILENAME = 'routes.txt';
+const DIST_MAPPER_JS_FILE = 'src/functions/utils/case-sensitive-video-id-mapper.js';
+console.log('Generating routes.txt for dynamic routes pre rendering');
+
+const getDistFilename = () => {
+  return DIST_DYNAMIC_ROUTES_FILENAME;
+}
+
+const getMapperFile = () => {
+  return DIST_MAPPER_JS_FILE;
+}
+
+let videoIds = [];
+
+/**
+ * Fetch the playlist by id and map the response to return only the ids
+ * @returns {Promise<void>}
+ */
+const getVideoIds = async () => {
+  // If video ids already computed return them
+  if(videoIds && videoIds.length > 0) {
+    return new Promise(resolve => resolve(videoIds));
+  }
+  // Fetch the playlist of ngx-darija
+  return fetch('http://localhost:8889/.netlify/functions/playlist')
+    .then(res => res.json())
+    .then(playlist => {
+      if (playlist && playlist.error) {
+        throw new Error(`Error when fetching playlist: ${playlist.error.message}`);
+      } else {
+        const items = playlist.items;
+        console.log(`Got ${items.length} videos`);
+        videoIds = items.map(item => item.snippet.resourceId.videoId);
+        return  videoIds;
+      }
+    });
+}
+
+/**
+ * Resolves with the content need for the `routes.txt` file.
+ * =>
+ * ``/sessions/34dwe
+ * ``/sessions/234dk2
+ * @returns {Promise<string>}
+ */
+const getDynamicRoutesFileContent = async () => {
+// Get video ids
+  return dynamicRoutesGenerator.getVideoIds().then(videoIds => {
+    let fileContent = '';
+    // For each video append the dynamic route to the file content variable
+    videoIds.forEach(videoId => {
+      fileContent += `/sessions/${videoId}\n`;
+    });
+    return fileContent;
+  });
+}
+
+/**
+ * Computes and writes the `routes.txt` file
+ */
+const generateDynamicRoutesFile = () => {
+// Get file content
+  dynamicRoutesGenerator.getDynamicRoutesFileContent().then(fileContent => {
+    // Ensure file existing or create it
+    ensureFileSync(dynamicRoutesGenerator.getDistFilename());
+    // Write the file content
+    writeFileSync(dynamicRoutesGenerator.getDistFilename(), fileContent);
+    dynamicRoutesGenerator.generateYoutubeVideoIdCaseSensitiveMapper();
+  })
+    .catch(err => {
+      console.error('Error when generating dynamic routes file', err);
+      process.exit(1)
+    })
+}
+
+/**
+ * Generates a mapper js file that exports a function that converts lower case youtube video IDs
+ * to their matching case sensitive IDs
+ */
+const generateYoutubeVideoIdCaseSensitiveMapper = () => {
+  let mapperContent = 'const getCaseSensitiveYoutubeVideoId = (videoId) => {\n' +
```

Comment:
Wouldn't be better to have a hashmap (plain json) generate in a .json file required and read by the getCaseSensitiveYoutubeVideoId function? Also just a suggestion :)

Reply:
Yes way better I don't know why I am creating js function instead of json :D